room.cpp

```diff
@@ -163,9 +163,17 @@ void Room::setJoinState(JoinState state)
     emit joinStateChanged(oldState, state);
 }
 
+void Room::setLastReadEvent(User* user, QString eventId)
```

Comment:
Hmm, why is this a public function? Imho it is an implementation detail and might be easily confused with markMessageAsRead, therefore I'd move it to Room::Private.